src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,79 @@
+package duke.tasks;
+
+import duke.storage.Storage;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TaskList {
+    private static final ArrayList<Task> taskList = new ArrayList<>();
+
+    static String countTasks() {
+        String pluralSuffix = taskList.size() == 1 ? "" : "s";
+        return String.format("%s task%s", taskList.size(), pluralSuffix);
+    }
+
+    static String add(Task task) {
+        taskList.add(task);
+
+        return String.format("Great, have added the following task for you:\n%s\nNow you have %s.", task, countTasks());
+    }
+
+    static Task get(int index) {
+        return taskList.get(index - 1);
+    }
+
+    static String list() {
+        int task_no = 1;
+        StringBuilder prompt = new StringBuilder();
+
+        prompt.append(String.format("You have %s in the list\n", countTasks()));
+
+        for (Task t : taskList) {
+            prompt.append(String.format("%d. %s\n", task_no++, t.toString()));
+        }
+        return prompt.toString().trim();
+    }
+
+    static void save() {
+        int task_no = 1;
+        StringBuilder currentListState = new StringBuilder();
+
+        for (Task t : taskList) {
+            currentListState.append(String.format("%d. %s\n", task_no++, t.toString()));
+        }
+
+        Storage.save(currentListState.toString());
+    }
+
+    static Task delete(int index) {
+        Task deletedTask = taskList.remove(index - 1);
+        return deletedTask;
+    }
+
+    static void reinitialise() {
+        taskList.clear();
+    }
+
+    public static String find(String taskKeyword) {
+        List<Task> filteredTasks = taskList.stream()
+                .filter(
+                        task -> Arrays.stream(task
+                                .toString()
+                                .toLowerCase()
+                                .split(" "))
+                                .anyMatch(word -> word.equals(taskKeyword)))
+                .collect(Collectors.toList());
```

Comment:
I really like this approach! Maybe you can modifying it to (unless I'm misunderstanding what you're attempting to do here):

```suggestion
        List<Task> filteredTasks = taskList
                .stream()
                .filter(task -> task.contains(taskKeyword)
                .collect(Collectors.toList());
```

Lambdas weren't covered in the convention but my TA said to refer to [Google's](https://google.github.io/styleguide/javaguide.html). Typically there is a linebreak between each lambda operation

Reply:
Hey, @yusufaine going to take the other suggestions under advisement, not replying individually to all, but thanks, great observations! 

With regards to this particular comment, I wanted to root out false flags. 
Let the `tasklist` be `[borrow book, borrow book, booking meetings]`
In my approach, `find("book")` returns `borrow book` and `borrow book` but not `booking meetings`. I thought of this while mistakenly triggering a search with the query `b`, wasn't pretty. 

Hope that helps.